sdk/src/main/java/com/uid2/UID2Manager.kt

```diff
@@ -161,6 +166,42 @@ public class UID2Manager internal constructor(
         }
     }
 
+    /**
+     * Generates a new identity.
+     *
+     * Once set, assuming it's valid, it will be monitored so that we automatically refresh the token(s) when required.
+     * This will also be persisted locally, so that when the application re-launches, we reload this Identity.
+     *
+     * @throws InputValidationException Thrown if the given [IdentityRequest] is not valid. For a
+     * [IdentityRequest.Phone] we expect the given number to conform to the ITU E.164 Standard
+     * (https://en.wikipedia.org/wiki/E.164).
+     */
+    @Throws(InputValidationException::class)
+    public fun generateIdentity(identityRequest: IdentityRequest): Unit = afterInitialized {
```

Comment:
Does the caller need to know when the generation finished? At the moment this launches a coroutine so the caller will never know.

Reply:
Yeah I'd add an `onResult: (Result) -> Unit` parameter. `Result` being a simple sealed class with Error, Success, etc